scripts/distributeVersion.ts

```diff
@@ -0,0 +1,170 @@
+    }
+
+}
+
+function getVersionSync(): string {
+    try {
+        const packageJson: PackageJson = JSON.parse(fs.readFileSync(versionFilePath, "utf8"));
+        return packageJson.version;
+    }
+    catch (error) {
+        throw new Error("Error in getVersionSync: " + error);
+    }
+}
+
+function distributeVersion() {
+    const version = getVersionSync();
+
+    for (const filePath in formats) {
+        if (!fs.existsSync(filePath)) {
+            continue;
+        }
+        try {
+            const fileInfo = getFileInformationSync(filePath);
+            const format = formats[filePath];
+            fs.writeFile(fileInfo.filePath, fileInfo.content.replace(format.pattern,
+                format.serialize(version)));
+        }
+        catch (err) {
+            console.log(err);
+        }
+    }
+}
+
+function distributeNightlyVersion() {
+    function computeNightlyVersion(versionString: string): string {
+        // If the version string already contains "-nightly",
+        // then get the base string and update based on that.
+        const dashNightlyPos = versionString.indexOf("-dev");
+        if (dashNightlyPos >= 0) {
+            versionString = versionString.slice(0, dashNightlyPos);
+        }
+
+        // We're going to append a representation of the current time at the end of the current version.
+        // String.prototype.toISOString() returns a 24-character string formatted as 'YYYY-MM-DDTHH:mm:ss.sssZ',
+        // but we'd prefer to just remove separators and limit ourselves to YYYYMMDD.
+        // UTC time will always be implicit here.
+        const now = new Date();
+        const timeStr = now.toISOString().replace(/:|T|\.|-/g, "").slice(0, 8);
+
+        return `${versionString}-dev.${timeStr}`;
+    }
+
+    const nightlyVersion = computeNightlyVersion(getVersionSync());
+
+    for (const filePath in formats) {
+        if (!fs.existsSync(filePath)) {
+            continue;
+        }
+        const format = formats[filePath];
+        // No nightly-serialization is defined
+        if (!format.nightlySerialize) {
+            continue;
+        }
+        try {
+            const fileInfo = getFileInformationSync(filePath);
+
+            fs.writeFile(fileInfo.filePath, fileInfo.content.replace(format.pattern,
+                                            format.nightlySerialize(nightlyVersion)));
+        }
+        catch (err) {
+            console.log(err);
+        }
+    }
+}
+
+function main() {
+    let  distributeNightly = false;
+
+    // Parse command line input checking if the users specify whether to configure version for nightly publishing
+    if (process.argv.length > 2) {
+        // Parse command line arguments
+        const arg0 = process.argv[2];
+
+        if (arg0 === "-h" || arg0 === "--help") {
+            // display help and exit
+            console.log("Usage: ");
+            console.log("\tnode distributeVersion.js (--nightly || -n)");
+            return;

```

Comment:
You should also display help if no arguments were supplied. Abstract this out to a `displayUsage` function.